File: LAB7/ejercicioPython/modeloCuadratico.py
"""
Algebra Lineal Lab-7
Entrega: 10/7/20
@steven_wilson
"""

import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import matplotlib
matplotlib.use('TkAgg')

# Position - Level vs Salary
df = pd.read_excel("ejercicio/Data.xlsx", sheet_name='Mcuadratico')

x = ((df.iloc[:, 1]).to_numpy())
y = ((df.iloc[:, 2]).to_numpy())

p = np.polyfit(x, y, 2)

print('Position - Cuadratic Model\n', p)


# Valores de y calculados del ajuste
y_ajuste = p[0]*x*x + p[1]*x + p[2]

# Dibujamos los datos experimentales
p_datos, = plt.plot(x, y, 'b.')
# Dibujamos la recta de ajuste
p_ajuste, = plt.plot(x, y_ajuste, 'r-')

plt.title('Parabola de mejor ajuste')

plt.xlabel('Eje X')
plt.ylabel('Eje Y')

plt.legend(('Datos experimentales', 'Ajuste lineal'), loc="upper left")
plt.show()

File: LAB7/ejercicioR/modeloCuadratico.r
# Algebra Lineal - Lab7
# Entrega: 10/7/20
# @steven_wilson

library(dplyr)
library(ggplot2)
library(readxl)
library(pracma)

#obtener datos
datos <- read_excel("Data.xlsx",sheet="Mcuadratico")

#ingresar arrays a funcion polyfit
p <- polyfit(datos$Level, datos$Salary, 2)

#mostrar coeficientes
p

#graficar data
plot(datos$Level, datos$Salary, pch=16, cex.lab = 1.3, col = "red", xlab="x", ylab="y", main="Parabola de mejor ajuste")

#graficar recta de ajuste
yf <- polyval(p, datos$Level)
lines(datos$Level, yf, col="blue")